### DISCUSSION TITLE
Generate GraphQL tokens from CLI

### DISCUSSION THREAD

**Discussion by membla** [craftcms/cms#9930] [Ideas] [answered]
In order to better support automated CI/CD it would be very desirable to be able to create tokens for specific schemas using the CLI. As of today this can only be done through the control panel and as project config does not include tokens this makes deploying a GraphQL token/schema enabled setup require a manual step.

**Comment by brandonkelly** [comment, marked as answer]
Good idea! Just released Craft 3.7.15 with a new `graphql/create-token` command. There’s also a new `graphql/list-schemas` command to make it easy to discover the appropriate schema UID to use with it.

**Reply by membla** [reply]
Thank you!

**Reply by sjcallender** [reply]
Nice. Just a few days after we created a content migration for it. 😂
